cartographer/io/probability_grid_points_processor.cc

```diff
@@ -97,16 +132,25 @@ void ProbabilityGridPointsProcessor::Process(
 }
 
 PointsProcessor::FlushResult ProbabilityGridPointsProcessor::Flush() {
-  Eigen::Array2i offset;
-  std::unique_ptr<Image> image =
-      DrawProbabilityGrid(probability_grid_, &offset);
-  if (image != nullptr) {
-    if (draw_trajectories_ ==
-        ProbabilityGridPointsProcessor::DrawTrajectories::kYes) {
-      DrawTrajectoriesIntoImage(probability_grid_, offset, trajectories_,
-                                image->GetCairoSurface().get());
+  if (output_type_ == OutputType::kPng) {
+    Eigen::Array2i offset;
+    std::unique_ptr<Image> image =
+        DrawProbabilityGrid(probability_grid_, &offset);
+    if (image != nullptr) {
+      if (draw_trajectories_ ==
+          ProbabilityGridPointsProcessor::DrawTrajectories::kYes) {
+        DrawTrajectoriesIntoImage(probability_grid_, offset, trajectories_,
+                                  image->GetCairoSurface().get());
+      }
+      image->WritePng(file_writer_.get());
+      CHECK(file_writer_->Close());
     }
-    image->WritePng(file_writer_.get());
+  } else {  // OutputType::kPb
```

Comment:
Code is better than comments. Make this an `else if` and `LOG(FATAL)` in another else branch?

Reply:
Ok, done!